samples/bluetooth/hci_usb/README.rst

```diff
@@ -21,3 +22,28 @@ This sample can be found under :zephyr_file:`samples/bluetooth/hci_usb` in the
 Zephyr tree.
 
 See :ref:`bluetooth samples section <bluetooth-samples>` for details.
+
+Notes on HCI over USB
+*********************
+
+USB is a serial protocol. However, the implementation of the peripheral can
+expose multiple endpoints, in effect re-ordering serial data into per-endpoint
+receive queues.
+
+The Bluetooth specification defines separate endpoints per HCI packet type.
+
+As the HCI protocol was not originally designed with re-ordering in mind, using
+this transport may expose the application to deadlocks and general unexpected
+behavior.
+
+Because of this, the industry-standard way to expose a Bluetooth Controller over
+USB is now to use the H4 HCI transport over virtual serial (i.e. CDC class).
```

Comment:
"citation needed". I would really prefer to have some reference to official Bluetooth specification backing this.

Reply:
@cvinayak @carlescufi do you have any non-nordic sources?

Reply:
This is really not "industry-standard" @jori-nordic. Rather, this is something both Linux and Zephyr have implemented, but in fact many in the industry use UART instead of USB instead for Bluetooth. Please rephrase to describe the reality better.

Reply:
@jori-nordic I don't think we do expose any tty really, or are you talking about btproxy? That uses H4 headers to proxy controllers to a VM, but that doesn't mean the kernel driver is communicating using H4.

Reply:
what is the physical interface between one of those intel m.2 cards and the motherboard? I thought it was CDC, maybe I misremembered.